lib/internal/modules/esm/resolve.js

```diff
@@ -310,7 +310,8 @@ function finalizeResolution(resolved, base, preserveSymlinks) {
       }
     }
 
-    path = file;
+    resolved = file;
+    path = fileURLToPath(resolved);
```

Comment:
Why are you reassigning `resolved`? That variable is used again farther down in this function.
```suggestion
    path = fileURLToPath(file);
```

I’m still wary of this change. As it modifies variables that are used outside of the `if (getOptionValue('--experimental-specifier-resolution') === 'node')` scope, it could have unintended consequences for anyone passing that flag. Those unintended consequences would be further bugs imposed upon the modules team to fix.

Reply:
The bug is that `resolved` is *not* reassigned when it should be. If `preseveSymlinks` is passed in the function just returns `resolved` and never (meaningfully) looks at path

Reply:
Then please add some comments explaining this. The code by itself looks like a mistake.

Reply:
Added - but in fairness, the `!preserveSymlinks` block below *also* reassigns to `resovled`.